br/pkg/lightning/restore/check_info.go

```diff
@@ -399,535 +180,64 @@ func (rc *Controller) estimateSourceData(ctx context.Context) (int64, int64, boo
 }
 
 // localResource checks the local node has enough resources for this import when local backend enabled;
-func (rc *Controller) localResource(ctx context.Context, sourceSize int64) error {
+func (rc *Controller) localResource(ctx context.Context) error {
 	if rc.isSourceInLocal() {
-		sourceDir := strings.TrimPrefix(rc.cfg.Mydumper.SourceDir, storage.LocalURIPrefix)
-		same, err := common.SameDisk(sourceDir, rc.cfg.TikvImporter.SortedKVDir)
+		theChecker, err := rc.precheckItemBuilder.BuildPrecheckItem(CheckLocalDiskPlacement)
 		if err != nil {
 			return errors.Trace(err)
 		}
-		if same {
-			rc.checkTemplate.Collect(Warn, false,
-				fmt.Sprintf("sorted-kv-dir:%s and data-source-dir:%s are in the same disk, may slow down performance",
-					rc.cfg.TikvImporter.SortedKVDir, sourceDir))
-		}
-	}
-
-	storageSize, err := common.GetStorageSize(rc.cfg.TikvImporter.SortedKVDir)
-	if err != nil {
-		return errors.Trace(err)
-	}
-	localAvailable := int64(storageSize.Available)
-
-	var message string
-	var passed bool
-	switch {
-	case localAvailable > sourceSize:
-		message = fmt.Sprintf("local disk resources are rich, estimate sorted data size %s, local available is %s",
-			units.BytesSize(float64(sourceSize)), units.BytesSize(float64(localAvailable)))
-		passed = true
-	case int64(rc.cfg.TikvImporter.DiskQuota) > localAvailable:
-		message = fmt.Sprintf("local disk space may not enough to finish import, estimate sorted data size is %s,"+
-			" but local available is %s, please set `tikv-importer.disk-quota` to a smaller value than %s"+
-			" or change `mydumper.sorted-kv-dir` to another disk with enough space to finish imports",
-			units.BytesSize(float64(sourceSize)),
-			units.BytesSize(float64(localAvailable)), units.BytesSize(float64(localAvailable)))
-		passed = false
-		log.FromContext(ctx).Error(message)
-	default:
-		message = fmt.Sprintf("local disk space may not enough to finish import, "+
-			"estimate sorted data size is %s, but local available is %s,"+
-			"we will use disk-quota (size: %s) to finish imports, which may slow down import",
-			units.BytesSize(float64(sourceSize)),
-			units.BytesSize(float64(localAvailable)), units.BytesSize(float64(rc.cfg.TikvImporter.DiskQuota)))
-		passed = true
-		log.FromContext(ctx).Warn(message)
-	}
-	rc.checkTemplate.Collect(Critical, passed, message)
-	return nil
-}
-
-// CheckpointIsValid checks whether we can start this import with this checkpoint.
-func (rc *Controller) CheckpointIsValid(ctx context.Context, tableInfo *mydump.MDTableMeta) ([]string, bool) {
```

Comment:
The logic is extracted as `(*checkpointCheckItem).Check()`, in `br/pkg/lightning/restore/precheck_impl.go`, L540